### ISSUE TITLE
Native attribute submission for shadow <form> element

### ISSUE THREAD

**Issue by alim** [PolymerElements/iron-form#195]
### Description
First off let me say I am new to Polymer and JavaScript is not be my strong suit. So, hopefully, I am not too far off base here.

We are using the Rails MVC framework for an application we are building. We would like to use Polymer form input elements (e.g. **paper-input**) in the forms that we create. We are using Rails 5.1 with standard ERB templates that normally expect native HTML form elements to be used in them.

### Expected outcome
When creating a form in a Rails view and using **iron-form** with the `allow-redirect` attribute, we expect the following:

1. The `action` to match the URL path that is submitted in the `<form>`element.
2. The `method` to match the HTTP verb specified in the `<form>` element.
3. The `accept-charset` to match the **enctype** in the shadow `<form>` element.

For example, if the `<form>` element is specified as below:
```
 <form class="new_app_form" id="new_app_form" action="/en/apps" accept-charset="UTF-8" method="post">
 ```
 We would expect the `action` to be `/en/apps`, the `method` to be `POST` and the `accept-charset` to be UTF-8.

### Actual outcome
Independent of what is specified for the `action`, `method`, and `accept-charset` attributes of a `<form>` element, the values for those attributes *do not* get set. As a result, the `method`
attribute is *always* a `GET`, the `action` is set to URL used to display the form. The snippet from our application log is below:
```
Started GET "/apps/new?utf8=%E2%9C%93&app_form%5Bname%5D=&app_form%5Bgithub_url%5D=&app_form%5Bdescription%5D="
```

### Steps to reproduce
1. Create a Rail 5.1 application ERB template with a corresponding server endpoint for the form submission. The form template we used is below:
```
   <iron-form allow-redirect>
     <%= form_for(app, url: apps_path, method: :post) do |f| %>

       <paper-input
         label="<%= I18n.t('views.apps.create.app_name') %>"
         name="app_form[name]"
         value="<%= app.name %>">
       </paper-input>

       <paper-input
         label="<%= I18n.t('views.apps.create.github_url') %>"
         name="app_form[github_url]"
         value="<%= app.github_url %>">
       </paper-input>

       <paper-textarea
         label="<%= I18n.t('views.apps.create.description') %>"
         name="app_form[description]"
         value="<%= app.description %>">
       </paper-textarea>

       <turbine-form-footer>
         <button class="btn" slot="right">
           <%= I18n.t('components.turbine-form-footer.buttons.next') %>
         </button>
       </turbine-form-footer>

     <% end %>
   </iron-form>
```

2. The Rails framework hash some helper methods to create forms, but thethe native `<form>` element it produces is below:
```
   <form class="new_app_form" id="new_app_form" action="/en/apps" accept-charset="UTF-8" method="post">
    <input name="utf8" type="hidden" value="✓">
    <input type="hidden" name="authenticity_token" value="actual key removed">

    <paper-input label="Application Name" name="app_form[name]" value="" tabindex="0" aria-disabled="false">
    </paper-input>

    <paper-input label="Github URL" name="app_form[github_url]" value="" tabindex="0" aria-disabled="false">
    </paper-input>

    <paper-textarea label="Short Description" name="app_form[description]" value="" tabindex="0" aria-disabled="false">
    </paper-textarea>

    <turbine-form-footer>
      <button class="btn" slot="right">
        Next
      </button>
    </turbine-form-footer>

   </form>
```

3. Submit the form and our application receives a **GET** request as shown in the "Actual Results" section.

### Diagnosis and Proposed Fix
I think the issue is that in processing the form for a native element submission, the **iron-form** element does not copy the `method`, `action`, and `enctype` attributes into the shadow `<form>` element inside the **iron-form** element.  The snippet from the **iron-form** submit
function is shown below:
```
  // If we want a redirect, submit the form natively.
  if (this.allowRedirect) {
     // If we're submitting the form natively, then create a hidden element for
     // each of the values.
     for (element in json) {
       this.$.helper.appendChild(this._createHiddenElement(element, json[element]));
     }
     this.$.helper.submit();
     this.fire('iron-form-submit');
  } else {
     this._makeAjaxRequest(json);
  }
```

I would propose the following change:
```
  // If we want a redirect, submit the form natively.
  if (this.allowRedirect) {
    // If we're submitting the form natively, then create a hidden element for
    // each of the values.
    for (element in json) {
      this.$.helper.appendChild(this._createHiddenElement(element, json[element]));
    }

    // Copy the original form element attributes to the shadow form element
    this.$.helper.action = this._form.getAttribute('action');
    this.$.helper.method = this._form.getAttribute('method') || 'GET';
    this.$.helper.contentType = this._form.getAttribute('enctype');

    this.$.helper.submit();
    this.fire('iron-form-submit');
  } else {
    this._makeAjaxRequest(json);
  }
```
I could be way off base here and there may be a better way to solve the issue, but I thought I would at least make a suggestion.

### Browsers Affected
- [x] Chrome
- [x] Firefox
- [ ] Safari 9
- [ ] Safari 8
- [ ] Safari 7
- [ ] Edge
- [ ] IE 11
- [ ] IE 10

## Versions
* Polymer - 2.0.0
* webcomponents - 1.0.1
* Iron-form - 2.0-preview

**Comment by notwaldorf** [comment]
First of all, thank you so much for how you wrote this issue! All the research you did is really appreciated! This is a super straight forward fix (and you're correct that it's needed), so I'll add it to the branch in approximately 2 minutes!

**Comment by notwaldorf** [comment]
Fixed in 6df0d73

**Comment by alim** [comment]
Thanks for the quick response. It's greatly appreciated.
